### PR TITLE
Rework switcher for ephemeral mode to the dropdown with storage types

### PR DESCRIPTION
**Pull request by vparfonov**
### What does this PR do?
Replace `Temporary Storage` switcher 

with `Storage Type` selector

In case `Ephemeral` will be added:
```
attributes:
  persistVolumes: 'false'
```
In case `Asynchronous`:
```
attributes:
  persistVolumes: 'false'
  asyncPersist: 'true' 
```
If `Persistent` attribute not required 

How it looks like:
https://youtu.be/EU_4M1ZHELw

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/17230

The Che Server side is done in https://github.com/eclipse/che/issues/15960

### CODE REVIEW

File: src/app/index.module.ts [hunk 1]
```diff
@@ -278,7 +278,7 @@ initModule.config(['$routeProvider', ($routeProvider: che.route.IRouteProvider)
 
 }]);
 
-const DEV = false;
```
Comment (sleshchenko) [inline src/app/index.module.ts:281]: it should be reverted before merging, right?

Comment (akurinnoy) [reply src/app/index.module.ts:281]: yes, it shouldn't be merged

Comment (vparfonov) [reply src/app/index.module.ts:281]: yep

File: src/app/get-started/custom-workspace-tab/infrastructure-namespace-row/infrastructure-namespace-row.html [hunk 2]
```diff
@@ -27,7 +27,7 @@
 
     <!-- Two and More Namespaces -->
     <che-pf-select-single config="ctrl.namespaceSelect.config" value="ctrl.namespaceSelect.value"
-      on-select="ctrl.namespaceSelect.onSelect($value)" ng-if="ctrl.namespacesNumber>1">
+      on-select="ctrl.namespaceSelect.onSelect($value)" ng-if="ctrl.namespacesNumber">
```
Comment (olexii4) [inline src/app/get-started/custom-workspace-tab/infrastructure-namespace-row/infrastructure-namespace-row.html:30]: The comment:  ```<!-- Two and More Namespaces -->```
Is it correct ```ng-if="ctrl.namespacesNumber>1" -> ng-if="ctrl.namespacesNumber"```?

File: src/app/get-started/custom-workspace-tab/temporary-storage-row/temporary-storage-row.controller.ts [hunk 3]
```diff
@@ -26,58 +28,81 @@ export class TemporaryStorageRowController implements ng.IController, ITemporary
...
-    this.temporaryStorageSwitch = {
+    this.storageSelect = {
       config: {
-        name: 'temporaryStorage',
+        items: [STORAGE_TYPE.PERSISTENT.label, STORAGE_TYPE.EPHEMERAL.label, STORAGE_TYPE.ASYNCHRONOUS.label],
+        placeholder: 'Select a storage template'
+      },
+      value: STORAGE_TYPE.EPHEMERAL.label,
+      onSelect: value => {
+        console.log(value)
```
Comment (olexii4) [inline src/app/get-started/custom-workspace-tab/temporary-storage-row/temporary-storage-row.controller.ts:51]: Do we really need it ```  console.log(value) ```?

Comment (olexii4) [reply src/app/get-started/custom-workspace-tab/temporary-storage-row/temporary-storage-row.controller.ts:51]: I like the old one ```onChange```

In the case with ```onSelect```, it does not clear what will be if I select but don't change the value.
WDYT?

Comment (olexii4) [reply src/app/get-started/custom-workspace-tab/temporary-storage-row/temporary-storage-row.controller.ts:51]: in the case with ```onSelect``` and the same value, it shouldn't call  ```this.onChangeStorageType```
https://github.com/eclipse/che-dashboard/pull/58/files#diff-fbf64ab21e61829eed60e7ce864f9cb0R106

File: src/app/get-started/get-started-tab/get-started-tab.controller.ts [hunk 4]
```diff
@@ -117,13 +133,23 @@ export class GetStartedTabController {
     this.devfileRegistry.fetchDevfile(this.devfileRegistryUrl, selfLink)
       .then(() => {
         const devfile = this.devfileRegistry.getDevfile(this.devfileRegistryUrl, selfLink);
-        if (this.ephemeralMode) {
+        console.log('=======>>>>' + this.storageType);
```
Comment (olexii4) [inline src/app/get-started/get-started-tab/get-started-tab.controller.ts:136]: remove ```console.log('=======>>>>' + this.storageType);```

File: src/app/workspaces/workspace-details/workspace-overview/workspace-details-overview.controller.ts [hunk 5]
```diff
@@ -79,6 +80,8 @@ export class WorkspaceDetailsOverviewController {
   private isLoading: boolean;
   private isEphemeralMode: boolean;
   private attributes: che.IWorkspaceConfigAttributes;
+  private storageType: { label: any; id: number; description:string};
```
Comment (olexii4) [inline src/app/workspaces/workspace-details/workspace-overview/workspace-details-overview.controller.ts:83]: format ``` { label: any; id: number; description:string}```

File: src/app/workspaces/workspace-details/workspace-overview/workspace-details-overview.controller.ts [hunk 6]
```diff
@@ -142,11 +145,30 @@ export class WorkspaceDetailsOverviewController {
...
     this.attributesCopy = angular.copy(this.cheWorkspace.getWorkspaceDataManager().getAttributes(this.workspaceDetails));
 
     this.updateInfrastructureNamespace();
   }
 
+  getStorageType() {
+    if (!this.attributes) {
+      return STORAGE_TYPE.EPHEMERAL;
+    }
+    if (this.attributes.persistVolumes as string === 'true') {
```
Comment (olexii4) [inline src/app/workspaces/workspace-details/workspace-overview/workspace-details-overview.controller.ts:159]: Maybe ```this.attributes.persistVolumes as string === 'true'  -> this.attributes.persistVolumes == 'true'``` will be better.
WDYT?

File: src/app/workspaces/workspace-details/workspace-overview/workspace-details-overview.controller.ts [hunk 7]
```diff
@@ -361,6 +383,52 @@ export class WorkspaceDetailsOverviewController {
...
+
+  updateStorageType() {
+    switch (this.storageType.id) {
+        case STORAGE_TYPE.PERSISTENT.id: {
+          this.storageDescription = STORAGE_TYPE.PERSISTENT.description;
+          if (!this.attributesCopy) {
+            this.attributes = undefined;
+          } else {
+            if ((this.attributesCopy.persistVolumes as string) === 'true') {
+              (this.attributes.persistVolumes as string) = 'true';
```
Comment (olexii4) [inline src/app/workspaces/workspace-details/workspace-overview/workspace-details-overview.controller.ts:398]:
```if ((this.attributesCopy.persistVolumes as string) === 'true') {
              (this.attributes.persistVolumes as string) = 'true';
....```
Is it copy/paste error?

File: src/app/get-started/custom-workspace-tab/infrastructure-namespace-row/infrastructure-namespace-row.html (hunk 2 above)
Comment (akurinnoy) [reply src/app/get-started/custom-workspace-tab/infrastructure-namespace-row/infrastructure-namespace-row.html:30]: you're right, it was changed accidentally

File: src/components/typings/che.d.ts [hunk 8]
```diff
@@ -343,6 +343,7 @@ declare namespace che {
 
   export interface IWorkspaceConfigAttributes {
     persistVolumes?: 'false' | 'true'; // explicitly indicates turning the ephemeral mode on
+    asyncPersist?:'false' | 'true'; // explicitly indicates turning the asynchronus storage mode on
```
Comment (olexii4) [inline src/components/typings/che.d.ts:346]: formatting ```asyncPersist?:'false' ```

**Comment by akurinnoy** [comment]
### Screenshots

<details>
<summary>Custom Workspace Tab</summary>

Screenshot 2020-07-14 at 10 55 45

Screenshot 2020-07-14 at 10 55 50

Screenshot 2020-07-14 at 10 55 56

</details>

<details>
<summary>Workspace Details Overview Tab</summary>

Screenshot 2020-07-14 at 10 56 09

Screenshot 2020-07-14 at 10 56 15

Screenshot 2020-07-14 at 10 56 41

</details>

File: src/components/che-pf-widget/popover/che-pf-popover.styl [hunk 9]
```diff
@@ -0,0 +1,10 @@
+.che-pf-popover
+  display: inline-block;
```
Comment (olexii4) [inline src/components/che-pf-widget/popover/che-pf-popover.styl:2]:
```
.che-pf-popover
  display inline-block
  position relative

...
```

**Comment by olexii4** [comment]
It didn't work for me
Screenshot from 2020-07-24 15-37-46
Screenshot from 2020-07-24 15-38-52

**Comment by akurinnoy** [comment]
@olexii4 it's expected because openshift.io doesn't support this feature yet. Try to test it using nightly Che as a target.

**Comment by olexii4** [comment]
LGTM

**Comment by olexii4** [comment]
Screenshot from 2020-07-27 09-40-42
Screenshot from 2020-07-27 09-41-36

**Comment by olexii4** [comment]
Tested with  v7.17.0

**Pull request merged**